clang/lib/AST/Decl.cpp

```diff
@@ -5569,7 +5569,8 @@ FunctionDecl *FunctionDecl::CreateDeserialized(ASTContext &C, GlobalDeclID ID) {
 
 bool FunctionDecl::isReferenceableKernel() const {
   return hasAttr<CUDAGlobalAttr>() ||
-         DeviceKernelAttr::isOpenCLSpelling(getAttr<DeviceKernelAttr>()) || hasAttr<DeviceKernelAttr>();
+         DeviceKernelAttr::isOpenCLSpelling(getAttr<DeviceKernelAttr>()) ||
+         hasAttr<DeviceKernelAttr>() || hasAttr<SYCLKernelAttr>();
```

Comment:
I'm not sure about this. `isReferenceableKernel()` has no comments that explain what it is intended for, but I wouldn't expect functions annotated with `sycl_kernel` to be "referenceable kernels" in common parlance.